src/item.cpp

```diff
@@ -2755,7 +2768,7 @@ std::set<matec_id> item::get_techniques() const
 
 bool item::goes_bad() const
 {
-    return is_food() && type->comestible->spoils;
+    return is_food() && !has_flag( "IRRADIATED" ) && type->comestible->spoils;
```

Comment:
I didn't check how often is `goes_bad` called. It might be worth looking at, since it is possible that it gets called a lot.

In such a case it would probably be a good idea to cache spoil protection as a `bool` or at least getting the "IRRADIATED" string out into a static constant.